### PR TITLE
[amd64] Obtain the trampoline argument in the generic_trampoline in the AOT case without decoding the call instruction.

### PR DESCRIPTION
**Pull request by vargaz**

### CODE REVIEW

File: mono/mini/aot-runtime.c [hunk 1]
```diff
@@ -5373,6 +5375,11 @@ load_function_full (MonoAotModule *amodule, const char *name, MonoTrampInfo **ou
 			if (ji->type == MONO_PATCH_INFO_SPECIFIC_TRAMPOLINE_LAZY_FETCH_ADDR) {
```
Comment (jaykrell) [inline mono/mini/aot-runtime.c]: Should probably be a switch. I was going to do that months ago but limited the churn (when I was removing nearby hashtable use).

**Comment by vargaz** [comment]
@monojenkins build failed

**Comment by vargaz** [comment]
@monojenkins build failed

**Comment by vargaz** [comment]
@monojenkins build failed

**Comment by vargaz** [comment]
@monojenkins build failed

**Pull request merged**